services/wallet/router/fees/estimated_time.go

```diff
@@ -102,14 +115,165 @@ func (f *FeeManager) estimatedTime(feeHistory *FeeHistory, maxFeePerGas *big.Int
 	return MoreThanFiveMinutes
 }
 
-func (f *FeeManager) getFeeHistorySorted(feeHistory *FeeHistory) ([]*big.Int, error) {
-	fees := []*big.Int{}
-	for _, fee := range feeHistory.BaseFeePerGas {
-		i := new(big.Int)
-		i.SetString(strings.Replace(fee, "0x", "", 1), 16)
-		fees = append(fees, i)
+func (f *FeeManager) convertToBigIntAndSort(hexArray []string) []*big.Int {
+	values := []*big.Int{}
+	for _, sValue := range hexArray {
+		iValue := new(big.Int)
+		_, ok := iValue.SetString(sValue, 0)
+		if !ok {
+			continue
+		}
+		values = append(values, iValue)
+	}
+
+	sort.Slice(values, func(i, j int) bool { return values[i].Cmp(values[j]) < 0 })
+	return values
+}
+
+func (f *FeeManager) getFeeHistoryForTimeEstimation(ctx context.Context, chainID uint64) (*FeeHistory, error) {
+	blockCount := uint64(10) // use the last 10 blocks for L1 chains
+	if chainID != common.EthereumMainnet &&
+		chainID != common.EthereumSepolia &&
+		chainID != common.AnvilMainnet {
+		blockCount = 50 // use the last 50 blocks for L2 chains
+	}
+	return f.getFeeHistory(ctx, chainID, blockCount, "latest", []int{RewardPercentiles2})
+}
+
+func calculateTimeForInclusion(chainID uint64, expectedInclusionInBlock int) uint {
+	blockCreationTime := common.GetBlockCreationTimeForChain(chainID)
+	blockCreationTimeInSeconds := uint(blockCreationTime.Seconds())
+
+	expectedInclusionTime := uint(expectedInclusionInBlock) * blockCreationTimeInSeconds
```

Comment:
We can get rid of all constants by using simpler approach:
```go
if expectedInclusionTime > 60 {
	return EstimatedTimeMoreThan1Min
}
	
return (expectedInclusionTime / 5) * 5
```

Reply:
@vkjr no, it's not like that. Different chains have different block creation times and we have predefined times (you can see in the description of this PR) that we're returning so we need to evaluate where, in which ranged the `expectedInclusionTime := uint(expectedInclusionInBlock) * blockCreationTimeInSeconds` falls and return that value.